syntax = "proto3";

package module_a;
import "generator_test/module_b.proto";

enum BasicEnum {
    UNKNOWN = 0;
    VALUE_1 = 1;
    ABC = 2;
    lower_case_value = 10;
}

message BasicSubItem {
    string name = 1;
    SomePoint singlePoint = 2;
}
message SomePoint {
    float x = 1;
    float y = 2;
}

message BasicMessageA {
    string name = 1;
    int32 int_value = 2;
    float float_value = 3;
    bool bool_value = 4;
    BasicEnum enum_value = 5;
    repeated int32 repeated_field = 6;
    BasicSubItem sub_item = 7;

    optional string o_name = 51;
    optional int32 o_int_value = 52;
    optional float o_float_value = 53;
    optional bool o_bool_value = 54;
    optional BasicEnum o_enum_value = 55;
}

message MsgWithRepeatedProps {
    string txt = 1;
    repeated string lines = 2;
    repeated BasicEnum enums = 3;
    repeated SomePoint points = 4;
}


message MsgWithOneOfProps {
    string txt = 1;
    oneof pointAorB {
        SomePoint point_a = 10;
        SomePoint point_b = 11;
    }
}

message MsgWithSet {
    module_b.ItemB item = 1;
    map<string, string> a_string_map = 2;
    map<string, module_b.ItemB> items_map = 3;
}